std-bits/table/src/main/java/org/enso/table/data/column/operation/unary/IsNaNOperation.java

```diff
@@ -21,27 +25,51 @@ public boolean canApply(ColumnStorage storage) {
   }
 
   @Override
-  protected void applyLong(ColumnLongStorage longStorage, Builder builder, MapOperationProblemAggregator problemAggregator) {
-    var boolBuilder = (BoolBuilder)builder;
-    UnaryOperation.applyOverLongStorage(longStorage, true, builder, (isNothing, value) -> boolBuilder.appendBoolean(false));
+  public ColumnStorage apply(
+      ColumnStorage storage, MapOperationProblemAggregator problemAggregator) {
+    if (storage instanceof ColumnLongStorage
+        && storage instanceof ColumnStorageWithNothingMap withNothingMap) {
+      // For a Column of Longs where we have the Nothing map, we can produce result immediately.
+      return new BoolStorage(
+          new BitSet(), withNothingMap.getIsNothingMap(), (int) storage.getSize(), false);
```

Comment:
a bit nitpick-y but maybe we could have a `BoolStorage::allFalse()` helper constructor, since we are using this pattern in multiple places and also that would make it much easier to understand what is happening.

E.g. exposing the `negated = false` bit here is mixing the layers of abstraction a bit too much IMO (that is an implementation detail of `BoolStorage`)